openhands/runtime/impl/remote/remote_runtime.py

```diff
@@ -361,49 +325,47 @@ def run_action(self, action: Action) -> Observation:
                 return NullObservation('')
             action_type = action.action  # type: ignore[attr-defined]
             if action_type not in ACTION_TYPE_TO_CLASS:
-                return FatalErrorObservation(
-                    f'[Runtime (ID={self.runtime_id})] Action {action_type} does not exist.'
-                )
+                raise ValueError(f'Action {action_type} does not exist.')
             if not hasattr(self, action_type):
-                return FatalErrorObservation(
-                    f'[Runtime (ID={self.runtime_id})] Action {action_type} is not supported in the current runtime.'
+                return ErrorObservation(
```

Comment:
I thought about this one, but went back on it. If we're in an alternative runtime, it seems reasonable to report this back to the agent so it can try another strategy. I'm not sure we can expect every Agent to handle ahead of time what the runtime capabilities are